crypto/hpke/hpke_util.c

```diff
@@ -211,12 +213,12 @@ const OSSL_HPKE_KDF_INFO *ossl_HPKE_KDF_INFO_find_id(uint16_t kdfid)
 
 const OSSL_HPKE_KDF_INFO *ossl_HPKE_KDF_INFO_find_random(OSSL_LIB_CTX *ctx)
 {
-    unsigned char rval = 0;
-    int sz = OSSL_NELEM(hpke_kdf_tab);
+    uint32_t rval = 0;
+    int err = 0;
+    size_t sz = OSSL_NELEM(hpke_kdf_tab);
 
-    if (RAND_bytes_ex(ctx, &rval, sizeof(rval), 0) <= 0)
-        return NULL;
-    return &hpke_kdf_tab[rval % sz];
+    rval = ossl_rand_uniform_uint32(ctx, sz, &err);
```

Comment:
@paulidale Why have you written this API like this?

Reply:
I don't understand?

Reply:
Guessing: I guess the question is why not do ``err =  ossl_rand_uniform_uint32(ctx, sz, &rval)`` instead?

Reply:
I followed the pattern in `include/internal/safe_math.h` which was where I originally intended this to go.
The main idea behind this pattern is that a number of operations can be done in succession and the error checked for at the end.

For example, 
```c
x = ossl_rand_uniform_uint32(ctx, range, &err);
x = safe_add_uint32(x, lower_bound, &err);

if (err) ...
```

Reply:
That is a pattern that requires err to be initialised. If the intention is at some point to make the API public, then it probably should be changed.

Reply:
This works for me:

```c
int OSSL_RAND_uniform_uint32(OSSL_LIB_CTX *ctx, uint32_t range, uint32_ *rnd)
{
    int err = 0;

    *rnd = ossl_rand_uniform_uint32(ctx, range, &err);
    return err;
}
```

Not that I've any intention of making this public at this stage.
It's also not going to be difficult to swap the API around when we want it public.

Reply:
'case it helps, I'm neutral, either pattern is fine by me.

Reply:
I'd keep it as is (at least for now).